openedx/core/djangoapps/user_authn/api/form_fields.py

```diff
@@ -163,3 +212,154 @@ def add_job_title_field():
+    # which allows the user to input the city in which they live.
+    city_label = _("City")
+
+    return {
+        'name': 'city',
+        'type': SUPPORTED_FIELDS_TYPES['TEXT'],
+        'label': city_label,
+        'error_message': accounts.REQUIRED_FIELD_CITY_MSG,
+    }
+
+
+def add_honor_code_field(separate_honor_and_tos):
+    """
+    Returns a honor code field to a description
+    """
+
+    # Separate terms of service and honor code checkboxes
+    if separate_honor_and_tos:
+        terms_label = _("Honor Code")
+        terms_link = marketing_link("HONOR")
+
+    # Combine terms of service and honor code checkboxes
+    else:
+        # Translators: This is a legal document users must agree to
+        # in order to register a new account.
+        terms_label = _("Terms of Service and Honor Code")
+        terms_link = marketing_link("HONOR")
+
+    # Translators: "Terms of Service" is a legal document users must agree to
+    # in order to register a new account.
+    label = Text(_(
+        "I agree to the {platform_name} {terms_of_service_link_start}{terms_of_service}{terms_of_service_link_end}"
+    )).format(
+        platform_name=configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME),
+        terms_of_service=terms_label,
+        terms_of_service_link_start=HTML("<a href='{terms_link}' rel='noopener' target='_blank'>").format(
+            terms_link=terms_link
+        ),
+        terms_of_service_link_end=HTML("</a>"),
+    )
+
+    # Translators: "Terms of Service" is a legal document users must agree to
+    # in order to register a new account.
+    error_msg = _("You must agree to the {platform_name} {terms_of_service}").format(
+        platform_name=configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME),
+        terms_of_service=terms_label
+    )
+    field_type = 'CHECKBOX'
+
+    if not separate_honor_and_tos:
+        field_type = 'TEXT'
+
+        pp_link = marketing_link("PRIVACY")
+        label = Text(_(
+            "By creating an account, you agree to the \
+              {terms_of_service_link_start}{terms_of_service}{terms_of_service_link_end} \
+              and you acknowledge that {platform_name} and each Member process your personal data in accordance \
+              with the {privacy_policy_link_start}Privacy Policy{privacy_policy_link_end}."
+        )).format(
+            platform_name=configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME),
+            terms_of_service=terms_label,
+            terms_of_service_link_start=HTML("<a href='{terms_url}' rel='noopener' target='_blank'>").format(
+                terms_url=terms_link
+            ),
+            terms_of_service_link_end=HTML("</a>"),
+            privacy_policy_link_start=HTML("<a href='{pp_url}' rel='noopener' target='_blank'>").format(
+                pp_url=pp_link
+            ),
+            privacy_policy_link_end=HTML("</a>"),
```

Comment:
Let's simplify it and send a simple label like "Honor Code" or "Terms of Service and Honor Code". We can then define the messages on frontend.